packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,294 @@
+}
+
+function Pagination({
+  className,
+  compact,
+  currentPage,
+  customUrl,
+  onPageChange,
+  leftLabel,
+  rightLabel,
+  totalPages,
+}: PaginationProps): React.ReactElement {
+  const classes = classNames('ds-c-pagination', className);
+
+  /**
+   * `useState` and `useEffect` determine if
+   * mobile layout of component is rendered.
+   */
+
+  const [isMobile, setIsMobile] = React.useState(false);
+  React.useEffect(() => {
+    // Mobile media query derived from: https://design.cms.gov/guidelines/responsive/
+    const media = window.matchMedia('(max-width: 543px)');
+    if (media.matches !== isMobile) {
+      setIsMobile(media.matches);
+    }
+    const listener = () => {
+      setIsMobile(media.matches);
+    };
+
+    media.addEventListener('change', listener);
+    return () => media.removeEventListener('change', listener);
+  }, [isMobile]);
+
+  const pageChange = React.useCallback(
+    (page) => (evt: React.MouseEvent) => onPageChange(evt, page),
+    [onPageChange]
+  );
+
+  const pages = [];
+
+  /**
+   * If `compact` or `isMobile` is true,
+   * don't run code to populate `pages[]`.
+   */
+  if (!compact || !isMobile) {
+    const pageRange = paginationBuilder(currentPage, totalPages);
+
+    if (pageRange[0] >= 2) {
+      /**
+       * If `pageRange` begins with a page of 2 or greater,
+       * begin Pagination with Page 1
+       */
+      pages.push(
+        <Page
+          customUrl={customUrl}
+          key="page-1"
+          index={1}
+          isActive={currentPage === 1}
+          onPageChange={pageChange(1)}
+        />
+      );
+
+      /**
+       * If `pageRange` doesn't equal 2, second Pagination element is Ellipses,
+       * otherwise page count continues.
+       */
+      if (pageRange[0] !== 2) {
+        pages.push(<Ellipses key="ellipses-1" />);
+      }
+    }
+
+    /**
+     * Renders all Page components in range (3 pages) to Pagination component.
+     */
+
+    pageRange.map((page) => {
+      pages.push(
+        <Page
+          customUrl={customUrl}
+          key={`page-${page}`}
+          index={page}
+          isActive={currentPage === page}
+          onPageChange={pageChange(page)}
+        />
+      );
+    });
+
+    /**
+     * Defines if/when the Ellipses component appears
+     * at the end of the Pagination component -
+     * as long as there are fewer than 7 pages.
+     */
+    if (currentPage <= totalPages - 3 && totalPages > maxVisiblePages) {
+      if (currentPage < totalPages - 3) {
+        pages.push(<Ellipses key="ellipses-2" />);
+      }
+
+      pages.push(
+        <Page
+          customUrl={customUrl}
+          key={`page-${totalPages}`}
+          index={totalPages}
+          isActive={currentPage === totalPages}
+          onPageChange={pageChange(totalPages)}
+        />
+      );
+    }
+  }
+
+  return (
+    <nav className={classes} role="navigation" aria-label="Pagination Navigation">
+      {currentPage !== 1 && (
+        <Button
```

Comment:
It looks like in the output, this is a link with a `role="button"`. Is that happening because of the button component? I seem to remember something related to that.

Reply:
I think it might be good to add a prop here for an `aria-label` because "Next" and "Previous" aren't specific enough in most cases. Maybe the default is "Next page" and "Previous page", but teams could change that as necessary.

Reply:
Re: `role="button"`, it does look like [that's baked into our Button component](https://github.com/CMSgov/design-system/blob/development/packages/design-system/src/components/Button/Button.tsx#L130). 

Good catch on the `aria-label` on nav elements!